src/evict/evict_lru.c

```diff
@@ -354,6 +354,33 @@ __evict_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
     return (ret);
 }
 
+/*
+ * __evict_set_saved_walk_tree --
+ *     Set saved walk tree maintaining use count. Call it with NULL to clear the saved walk tree.
+ */
+static void
+__evict_set_saved_walk_tree(WT_SESSION_IMPL *session, WT_DATA_HANDLE *new_dhandle)
+{
+    WT_CACHE *cache;
+    WT_DATA_HANDLE *saved_walk_tree;
+
+    cache = S2C(session)->cache;
+    saved_walk_tree = cache->walk_tree;
```

Comment:
Why we need this local variable? Can this change currently? If that's the case, should we define it as volatile to ensure this read is not optimized out?

Reply:
It's just for convenience so that I don't have to repeat `cache->walk_tree`

Reply:
ok, thanks. The `saved` working makes me think that it may change underneath.

Reply:
Ah or I'll rename it